src/terminal.rs

```diff
@@ -70,23 +76,52 @@ impl TerminalLoggerBuilder {
         self
     }
 
+    /// Sets KVFilter
+    pub fn kvfilter(&mut self,
+                    level: Severity,
+                    only_pass_any_on_all_keys: Option<KVFilterList>,
+                    always_suppress_any: Option<KVFilterList>) -> &mut Self {
+        self.kvfilterparameters = Some(KVFilterParameters {
+            severity: level,
+            only_pass_any_on_all_keys,
+            always_suppress_any
+        });
+        self
+    }
+
     fn build_with_drain<D>(&self, drain: D) -> Logger
     where
         D: Drain + Send + 'static,
         D::Err: Debug,
     {
-        let drain = Async::new(drain.fuse())
-            .chan_size(self.channel_size)
-            .build()
-            .fuse();
-        let drain = self.level.set_level_filter(drain).fuse();
-
-        match self.source_location {
-            SourceLocation::None => Logger::root(drain, o!()),
-            SourceLocation::ModuleAndLine => {
-                Logger::root(drain, o!("module" => FnValue(module_and_line)))
+        fn finalfuse<D>(pars: &TerminalLoggerBuilder, drain: D) -> Logger
+            where
+                D: Drain + Send + 'static,
+                D::Err: Debug {
+            let drain = Async::new(drain.fuse())
+                .chan_size(pars.channel_size)
+                .build()
+                .fuse();
+            let drain =
+                AtomicSwitch::new(pars.level.set_level_filter(drain).fuse());
```

Comment:
It seems that there is no way to obtain `AtomicSwitchCtrl` from the drain after creating the logger (note I have never used `AtomicSwitch`).

Reply:
I checked, there is ...

Reply:
It seems helpful to add an example code (rustdoc documentation) that explains the existence and an usage of `AtomicSwitch`.

Reply:
Yupp, you were correct. The way I've hidden the Atomic it's useless. I have to refactor that which is more complex than it looks due to Drain having so many requirements as trait to write generics over it. I have to solve the case and will get back once I have that done.